firewalldb/actions_sql.go

```diff
@@ -0,0 +1,429 @@
+package firewalldb
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/lightninglabs/lightning-terminal/accounts"
+	"github.com/lightninglabs/lightning-terminal/db"
+	"github.com/lightninglabs/lightning-terminal/db/sqlc"
+	"github.com/lightninglabs/lightning-terminal/session"
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/sqldb"
+)
+
+// SQLAccountQueries is a subset of the sqlc.Queries interface that can be used
+// to interact with the actions table.
+type SQLAccountQueries interface {
+	GetAccount(ctx context.Context, id int64) (sqlc.Account, error)
+	GetAccountIDByAlias(ctx context.Context, alias int64) (int64, error)
+}
+
+// SQLActionQueries is a subset of the sqlc.Queries interface that can be used
+// to interact with action related tables.
+//
+//nolint:lll
+type SQLActionQueries interface {
+	SQLSessionQueries
+	SQLAccountQueries
+
+	InsertAction(ctx context.Context, arg sqlc.InsertActionParams) (int64, error)
+	SetActionState(ctx context.Context, arg sqlc.SetActionStateParams) error
+	ListActions(ctx context.Context, arg sqlc.ListActionsParams) ([]sqlc.Action, error)
+	ListActionsPaginated(ctx context.Context, arg sqlc.ListActionsPaginatedParams) ([]sqlc.Action, error)
+	CountActions(ctx context.Context, arg sqlc.CountActionsParams) (int64, error)
+}
+
+// sqlActionLocator helps us find an action in the SQL DB.
+type sqlActionLocator struct {
+	// id is the DB level ID of the action.
+	id int64
+}
+
+func (s *sqlActionLocator) isActionLocator() {}
+
+// A compile-time check to ensure sqlActionLocator implements the ActionLocator
+// interface.
+var _ ActionLocator = (*sqlActionLocator)(nil)
+
+// GetActionsReadDB is a method on DB that constructs an ActionsReadDB.
+//
+// NOTE: This is part of the ActionDB interface.
+func (s *SQLDB) GetActionsReadDB(groupID session.ID,
+	featureName string) ActionsReadDB {
+
+	return &allActionsReadDB{
+		db:          s,
+		groupID:     groupID,
+		featureName: featureName,
+	}
+}
+
+// AddAction persists the given action to the database.
+//
+// NOTE: This is a part of the ActionDB interface.
+func (s *SQLDB) AddAction(ctx context.Context,
+	req *AddActionReq) (ActionLocator, error) {
+
+	var (
+		writeTxOpts db.QueriesTxOptions
+		locator     sqlActionLocator
+
+		actor          = sql.NullString{String: req.ActorName}
+		feature        = sql.NullString{String: req.FeatureName}
+		trigger        = sql.NullString{String: req.Trigger}
+		intent         = sql.NullString{String: req.Intent}
+		structuredJson = sql.NullString{String: req.StructuredJsonData}
+	)
+	if req.ActorName != "" {
+		actor.Valid = true
+	}
```

Comment:
Instead of these `if` cases here and below, we can instead just set:
`actor.Valid = opts.actorName != ""`

We can actually also just set during the `var` initialization, so that it's instead set as:
```
actor = sql.NullString{
	String: req.ActorName
	Valid: opts.actorName != ""
}
```